### ISSUE TITLE
context.setLineDash([0,0]) stop drawing operations

### ISSUE THREAD

**Issue by asturur** [Automattic/node-canvas#1055]
## Issue or Feature
When using ctx.setLineDash([0,0]), drawing calls done after are ignored.
While setting lineDash to [0,0] is not usefull, the same thing is tolerated by browser canvas.

Whenever this is a backend problem or not, node-canvas could inspect the dash setting for problematic sequences and ignore them in order to provide compatibility with canvas code.

As a side node, node-canvas looks like is tolerating ctx.setLineDash(array); where at least one element is not 0, for example [0, 3, 0] is OK or also [0, 0, 3] even if the resulting dash is slightly different from canvas at the line start.

## Steps to Reproduce
```js
const Canvas = require('canvas');
let { writeFileSync, createWriteStream } = require('fs');
let pngfile = createWriteStream("./bug.png");

const canvas = new Canvas(200, 200)
const ctx = canvas.getContext('2d')
ctx.setLineDash([0,0]);
ctx.fillStyle = 'green';
ctx.fillRect(10,10, 150,150);
ctx.lineWidth = 8;
ctx.strokeStyle = 'red';
ctx.fillStyle = 'blue';
ctx.strokeRect(10,10, 150,150);
ctx.fillRect(10,10, 150,150);
let stream = canvas.createPNGStream();

stream.on("data", function(chunk){
    pngfile.write(chunk);
});
stream.on("end", function(){
    console.log("saved png");
});
```

## Browser fiddle with same code
https://jsfiddle.net/yyg1adr3/

* Version of node-canvas (e.g. 1.6.X):
* Environment (e.g. node 6.9.5 on Mac OS X Sierra):

**Comment by asturur** [comment]
I should be able to provide a fix, i got the test suite running and rebuilding on my computer, i will give it a shot.

**Comment by asturur** [comment]
I made the PR on the 1.x branch since that is the stable one and the one i use, if is good i can cherrypick on master and open another pr for the current master.

Unless you prefer another strategy.

**Comment by Franthis** [comment]
I downloaded the latest v1.x and the test case provided here isn't working for me.  If I change `ctx.setLineDash([0,0]);` to be `ctx.setLineDash([1,1]);` it draws properly, though.  I also tried making the changes in https://github.com/Automattic/node-canvas/pull/1060 manually on top of 1.6.9 and that didn't work either.  Is there something else that I'm missing?  I believe the only relevant changes are the changes made to `src/CanvasRenderingContext2d.cc` in https://github.com/Automattic/node-canvas/pull/1060.
